### DISCUSSION TITLE
v2.0

### DISCUSSION THREAD

**Discussion by eduardofilo** [eduardofilo/RG350_adam_image#256] [Announcements]
## Installation instructions

* [In english](https://github.com/eduardofilo/RG350_adam_image/wiki/En:-2.-Install-&-Upgrade#fresh-installation): It is mandatory to follow the instructions or the console will not boot up.
* [En español](https://github.com/eduardofilo/RG350_adam_image/wiki/Es:-2.-Instalaci%C3%B3n-y-Actualizaci%C3%B3n#instalaci%C3%B3n-desde-cero): Es obligatorio seguir las instrucciones o la consola no arrancará.

## Upgrading instructions

This v2.0 contains an important change to facilitate upgrading to future versions. RetroArch savestates and standalone emulators that support them will now be stored on the external card, so they will not be lost when flashing the internal card with a new version of the image. The configuration files however have been kept on the internal card as they are part of the design of the image. Advanced users who make modifications to these configurations can still use Py Backup to backup/restore them.

To facilitate the special migration between the 1.x and 2.x branchs of the image, a small utility has been created that copies all the savestates to the new directory structure of the external card that will keep them from now on. SimpleMenu favorites and settings made at the ROM level of the emulator chosen to run them are also copied. The utility is the `adam_v2.0_migrator.opk` file which is included in the release files. The following video shows how to install and use this utility.

[![See video](https://img.youtube.com/vi/lMnhQhVmzKs/hqdefault.jpg)](https://www.youtube.com/watch?v=lMnhQhVmzKs "See video")

Once used, delete the `adam_v2.0_migrator.opk` OPK as it will no longer be needed.

## Change log

* First Adam release made with new build system. img file more compact and free of fragmentation.
* exFAT support enabled.
* All savestates now go to the EXT SD card, making it easier to update future releases of the image. Favorites and rom_preferences of SimpleMenu are also maintained in EXT SD card.
* ODbeta updated to 2022-04-03 (5.17 version) thanks to @pcercuei and all the collaborators of the [project](https://github.com/OpenDingux/linux).
* RetroArch updated to 2022-06-13 (1.10.3 version):
    * New official cores (only those in bold are accessible from SimpleMenu; for the rest use GMenu2X):
        * arduous_libretro.so: Arduboy
        * chailove_libretro.so: ChaiLove
        * gong_libretro.so: Pong
        * jumpnbump_libretro.so: Jump 'n Bump
        * lutro_libretro.so: Lua Engine (Lutro)
        * **retro8_libretro.so**: PICO-8. Used as default for PICO-8 in SimpleMenu.
        * vaporspec_libretro.so: VaporSpec
        * wasm4_libretro.so: WASM-4
        * xrick_libretro.so: Rick Dangerous
    * Cores previously taken from [Poligraf compilation](https://github.com/Poligraf/opendingux_ra_cores_unofficial) which have now become official (only those in bold are accessible from SimpleMenu; for the rest use GMenu2X):
        * 2048_libretro: 2048 port
        * 81_libretro.so: Sinclair ZX81
        * cannonball_libretro.so: Outrun
        * dinothawr_libretro: Dinothawr
        * **fuse_libretro.so**: Sinclair ZX Spectrum
        * gme_libretro.so: Game Music Emu
        * **gw_libretro.so**: Nintendo Game&Watch
        * nxengine_libretro: Cave Story
        * pocketcdg_libretro: Karaoke player
        * quasi88_libretro.so: PC-88
        * x1_libretro.so: Sharp X1
* Updated GMU music player with proper visualization in RG280 screen. Thanks to @plrguez (discussion #241).
* Fixed RG350_test. Now it shows battery level and CPU speed. Thanks to @a-dekker (discussion #254).

<hr /><em>This discussion was created from the release <a href='https://github.com/eduardofilo/RG350_adam_image/releases/tag/v2.0'>v2.0</a>.</em>

**Comment by a-dekker** [comment]
And for RG350M users that still have a very tiny fontsize in the GMU music player see this remark https://github.com/eduardofilo/RG350_adam_image/discussions/241#discussioncomment-2829485

**Reply by eduardofilo** [reply]
I have deleted that configuration file in this release to generate the most appropriate one for each screen on first launch as programmed by @plrguez. If the card is exchanged between different devices, then it is advisable to delete the one that will be generated the first time GMU is opened.

**Comment by K9-711** [comment]
Thanks for the new image. I just updated my RG350M. Not sure if I found a bug. 

I add a game to favorites.
Then I remove that game from favorites.
Now I want to add the game again to favorites. Press X and the device vibrates as it should.
But the game isn't in favorites. If I again try to add it the device doesn't vibrate anymore by pressing X.

**Reply by eduardofilo** [reply]
I have not been able to reproduce your problem, although I have been able to reproduce a similar problem that I imagine is related to yours. It is a SimpleMenu bug.

What I have seen is that inside the `favorites.sav` file, in some situations there is a residue in the fourth field if an entry from the APPS section has been added to favorites before (at least in my tests; there may be other conditions that produce this or other problems). In that case in the list of favorites, instead of the ROM name appears that residue. If the residue is an invisible character, it is possible that your ROM is actually in favorites but you don't see it. That's why you can't add it again. Manually edit the file on the external at the following path:

```simplemenu/favorites.sav```

Being now the external card in FAT32 or exFAT format, you can do it from the computer, although it can be dangerous depending on the editor you use if it alters the carriage return characters. If you edit the file, you will surely see your favourites ROMs. Try to fix what you see in the fourth field of the ROMs that you cannot see. The fields are separated by the ";" character.

**Reply by K9-711** [reply]
hmm I can't find anything. I attached the favorites.sav

Another thing is, I have Wonder Boy on the favorites. This game is listed in Simplemnu twice, and as onder Boy. So the W is missing. 

[favorites.zip](https://github.com/eduardofilo/RG350_adam_image/files/8933852/favorites.zip)

**Reply by K9-711** [reply]
I fixed it. I deleted the Wonder Boy line which was suspicious because of the missing W. Then I added it again within Simple Menu. Now adding, remove and adding again to favorites works perfect. It's not clear to me what actually caused the issue.

**Reply by piklz** [reply]
My error log from simplemenu says "critical error" just after loading favs file.. even if empty or not a symlink(using a real file in place of) it crashes simplemenu everytime I'm not sure why it's doing this 

So I fixed it kinda by removing favs file let it try again and then simple menu behaved and then it populated the folders for each console etc some games won't find the in game saves.. even though states are loading fine.. symlinks look ok so not sure what's changed

**Comment by SuperBadger77** [comment]
Hi, I formatted a 256GB card in exFAT with the intention on this release to migrate all my games on my 128GB card in FAT32 to the 256. I formatted the 256 in exFAT and flashed v2.0 to my 32GB internal card as I would noramlly. It appears the exFAT card isn't recognised? Normally if I put a blank card in and boot, the folder structure gets added. Nothing happened on the 256 - am I doing something wrong?

**Reply by eduardofilo** [reply]
I don't really have such a large card to test, but in all the tests I have done with smaller cards they have mounted correctly.

Try viewing the `/proc/mounts` file.

![screenshot002](https://user-images.githubusercontent.com/2923728/174483365-79f9bcef-82e9-4618-85bd-12930432f487.png)

As you can see in my case partition 1 (p1) of drive 1 (`/dev/mmcblk1` = the external card) is of exfat type and is mounted on `/media/sdcard`.

![screenshot003](https://user-images.githubusercontent.com/2923728/174483428-ce9ba0ac-362e-4d79-8155-6f703f5d075d.png)

Also make sure not to label the partition.

**Comment by Tobison** [comment]
I have a problem with migrator.opk on my RG350M. When I open this app, OpenDingux start and show "loading" but it can't finish. All the time I see OpenDingux logo with "loading...." subtitle. Could you help me?

**Reply by eduardofilo** [reply]
Where are you installing the OPK?
What version of Adam are you installing the OPK on?
Where are you running it from (GMenu2X or SimpleMenu)?

**Reply by eduardofilo** [reply]
I have tested on a RG350M installing the OPK in the `apps` folder of external card on Adam 1.4.2 and the situation you describe was reproduced by running the OPK from Commander. You have to run the OPK from GMenu2X or from SimpleMenu as shown in the video I indicated in the release update instructions:

https://www.youtube.com/watch?v=lMnhQhVmzKs

The only problem with this OPK is the same as all terminal applications in ODbeta. The screen boots to black and the terminal contents do not appear until you press a control (I recommend pressing one of the directions on the D-Pad). But this is not what you were describing so I understand that your problem was something else.

**Reply by Tobison** [reply]
Thank you for your response.  I took the card out of the console and put it in the computer and windows detected problems with the card which it fixed itself.  Then I uploaded the * .opk file again and after placing the card in the console, it was updated.  I only had a problem with the black screen you are writing about, but it was enough to press any button on the D-pad.  After that, the console updated with no problem.  Thank you very much for help.  I was installing Adam 2.0 on version 1.3.  and I was using Simple Menu.

**Reply by piklz** [reply]
same seems the ext sd card for me isnt mounting anymore(280v) if removed simple menu loads and warns no favs is available.. and at least shows the menu with default apps again

**Reply by piklz** [reply]
Since noticed that the normal (non simplemenu )menu loads the ext sdcard fine? But simple menu doesn't work bootloops and goes back to standard frontend ? 
I'm puzzled as the adma image v2 is working fine now and I don't know why ext sd wouldn't mount under simple menu and not it will if its the default frontend

**Reply by eduardofilo** [reply]
I'm not sure I understand your situation. I recommend that you analyze the situation in stages. Start with a clean install of the image on the internal card and with a clean external card other than the previous one and formatted in FAT32 (as exFAT support has only recently been introduced and is less tested). Unless you have problems with the contacts on the cards or with the quality of the cards themselves, the system should boot and create the directory structure on the external card. Then gradually copy what you had on your initial external card to the new one until you find what is causing the problems in SimpleMenu.

**Reply by piklz** [reply]
Seems that simple menu didn't like loading the ext sd that had previous fav items from(prev adam 1.x setup )  once I removed it(..and rebooted system could mount again?) and made new favs from simplemenu ,on next boot up it all was mounting fine ? Still not 100% sure why it happened but all my previous extsd stuff is working fine .

**Comment by Tobison** [comment]
I have another question.  Is it possible to connect the console to the TV via HDMI and be able to listen to the sounds from the TV?  I have an RG350M and only the image is transmitted via HDMI.  I use Adam 2.0.

**Reply by eduardofilo** [reply]
No, sorry. The Linux system installed in Adam does not support audio over HDMI yet.

**Comment by kaossino** [comment]
Hi, it's the first time I change the OS to my RG350.
I opened the RG350 and I removed the internal card, I formatted it and I flashed the Adam v2.0 image via Balena, I started the file and I selected the number 3, then I put the card back inside.
I formatted my second (external) SD card without a label and put it into my RG350 then turned it on.
I took the external SD and inside there were the folders created by Adam, I put the games inside each folder (eg NES rom games inside FC and images inside .previews), all the games I put were .zip.
I put the external SD back into RG350 and I only see the clock applications, file manager, etc.
I thought it was a bios problem and tried to figure out which ones I had to put, I put for example ATARI7800, Lynx, GB, GBC, I put the SD back into RG and I see RETROARCH but I can't see any games list.
I tried to extract the zip games of GB and FC and put the SD back in but I still see only RETROACRH and no games.

I do not know what to do. Hello and thanks

**Reply by eduardofilo** [reply]
Check FAQ number 10 because this could be your case.

https://github.com/eduardofilo/RG350_adam_image/wiki/En:-7.-FAQ#q10-why-dont-appear-in-simplemenu-the-roms-i-just-loaded-onto-the-external-card

**Reply by kaossino** [reply]
ok, in fact it has partially solved the problem, I only see the games that I have unzipped and they are no longer .zip, even if in the wiki it says that .zip works. Now I unzip everything

**Reply by eduardofilo** [reply]
SimpleMenu does nothing but filter extensions so ROMs in zip should appear if they are supported (as indicated in the table). Please give an example, i.e. system name and ROM filename (if possible attach screenshots or pictures of the directory containing the files, or terminal dumps by SSH).

**Reply by kaossino** [reply]
I extracted everything now, but first I had put the zip files of: atari7800, lynx, nes, snes, gg, sms, md, gb, gbc, gba, pce, 32x, neogeo, ngp, wonder

**Reply by kaossino** [reply]
Ok now I see everything, but I have problems with the images, in my folders the images I have are not called exactly like the roms and I have XML files that keep everything in order. Does Adam read XML files for image management?

**Reply by eduardofilo** [reply]
> Ok now I see everything, 

You mean you already see the zips too? What was the problem?

> but I have problems with the images, in my folders the images I have are not called exactly like the roms and I have XML files that keep everything in order. Does Adam read XML files for image management?

No. SimpleMenu is a bit... simpler than EmulationStation which is the one that can read those XMLs. The image names have to be named the same as the ROMs (except for the extension) as described here:
https://github.com/eduardofilo/RG350_adam_image/wiki/En:-3.-Content-installation#previews

It can be easily achieved by properly configuring skraper. This RetroGameCorps guide explains it very well:
https://www.youtube.com/watch?v=kdIdiQ-dO_Y&t=433s

**Reply by kaossino** [reply]
> You mean you already see the zips too? What was the problem?

No I have unzipped everything

> It can be easily achieved by properly configuring skraper. This RetroGameCorps guide explains it very well:

I had already tried to use that program precisely following the settings in that video, but it fails, it finds only a small part of the images.
I already have all the images, there is only one problem in the name.

**Reply by eduardofilo** [reply]
> No I have unzipped everything

Well, I assure you that ROMs in zip on systems that support that extension (practically all those that work on RetroArch) can be seen in SimpleMenu. If you are interested in finding the problem, send me the screenshots I asked for a few comments ago so I can help you.

**Reply by kaossino** [reply]
Ok, I wanted to do it all over again to see if it actually worked, this time it reads the zips, I have no idea what I could have done wrong before, but now it works and the zips have cleaner names so I was able to find the images via the software. Thanks a lot it's cool

**Comment by kaossino** [comment]
I figured out how to increase the brightness but instead the maximum volume is much lower than it was before with the official RG350 operating system, is there a way to increase it?

**Reply by eduardofilo** [reply]
Read FAQ number 7 that talks about something similar but in the opposite direction:

https://github.com/eduardofilo/RG350_adam_image/wiki/En:-7.-FAQ#q7-how-can-i-lower-the-minimum-volume-without-actually-muting-the-sound-the-minimum-is-still-very-high

**Reply by eduardofilo** [reply]
In your case, try to increase `Headphones` and `PCM` channels in alsamixer and/or `Volume Gain` in RetroArch.

**Reply by kaossino** [reply]
Ok thank you, work

**Reply by eduardofilo** [reply]
In the end, there was a problem. It will be solved (I hope) in v2.0.1. Check issue #263.

**Comment by ghost** [comment]
I have a lot of Chinese and Japanese games that I kept from when I bought the RG280v. I want to gift it to my coworker, who is Chinese.

Unfortunately, all the Chinese and Japanese game names show up as squares and gibberish symbols. Is there a way to install Chinese language on this? I'm not looking for Chinese menues or anything like that, just want the game names to be displayed properly.

**Reply by a-dekker** [reply]
Is this related https://github.com/eduardofilo/RG350_adam_image/discussions/129 ?

**Reply by ghost** [reply]
I think so. I'll have to figure that out then. Thanks!

**Reply by a-dekker** [reply]
There already is an issues created in the simplemenu repo for that https://github.com/fgl82/simplemenu/issues/194

Unfortunately the development of Simplemenu is stalled.

**Reply by ghost** [reply]
Thanks for the heads up on this! I wish it were as simple as installing a language pack.

**Reply by pcercuei** [reply]
GMenu2X should support Chinese though.

**Reply by ghost** [reply]
Is there a way for chinese to be installed onto the SD card in a way that would allow for proper viewing of game titles?

**Reply by a-dekker** [reply]
In most cases it will depend on how the application handles fonts. As Pcercuei mentioned, GMenu2X (de default GUI) can display Chinese characters. So not using Simplemenu and defaulting to GMenu2x could be a workaround.

**Reply by ghost** [reply]
> In most cases it will depend on how the application handles fonts. As Pcercuei mentioned, GMenu2X (de default GUI) can display Chinese characters. So not using Simplemenu and defaulting to GMenu2x could be a workaround.

How would one default to GMenu2x on Adam image? Would it require installing an additional app? Or is there a way to disable simple menu?

**Reply by a-dekker** [reply]
I made a proof of concept of a simplemenu version that adds support for more languages (incl. Chinese) for rom names. Be aware the fonts look a bit different than before, it is not polished.
Download https://github.com/a-dekker/repackaged-opks/raw/main/SimpleMenu-OD-BETA.opk and overwrite the existing file in /media/data/apps.

**Reply by neilswann80** [reply]
@a-dekker seriously nice work!  👍

**Reply by ghost** [reply]
@a-dekker Thanks  for this! I am having trouble finding the folder after I plug in the SD card. I checked INT1 and EXT2

**Reply by a-dekker** [reply]
it is the apps folder on TF1/INT

**Reply by ghost** [reply]
> it is the apps folder on TF1/INT

I can view these files, but I dont seem to have permission to read/write to it. (cant delete the old one and put on the new one). when I run it as its own independent app, the Proof of Concept one you provided works great! I just need to figure out how to overwrite the linux partition. (I run windows. I tried changing it within a virtual machine and with ext2explore

**Reply by a-dekker** [reply]
As I understand correctly you now want to update the adam image file itself? That's doable, but more complex. And not the goal of this exercise.
If we want to do something with this, it should be part of a regular Adam release, otherwise it will get lost on every update.
But as I said the simplemenu project seems dead, no new fixes or pull requests merged. So ultimately one would need a fork of simplemenu to continue changes I am afraid.

**Reply by ghost** [reply]
What I ended up doing was putting your version on the root of the INT1 SD card, and then using the Commander file manager app to move it and overwrite the existing simple menu app. Works perfectly and looks good! Thanks for making this!

I don’t plan on updating the image any time soon since I want to give this to a Chinese friend who is minimally tech savvy. So for her this device will be wysiwyg. Which is fine. Thank you again so much! I learned a lot about this system as a result!

**Reply by eduardofilo** [reply]
Great @a-dekker 
Which branch of the SimpleMenu repository did you use to make the modification the `main` or the `experimental` one? Do you think your modification is safe enough to include it in the 2.0.1 version of the image I am preparing now?
Thank you...

**Reply by a-dekker** [reply]
I used the experimental branch, the main branch seems to be very outdated.

My take would be: try to get a statement on the state of development of https://github.com/fgl82/simplemenu from fgl82.
If negative or no answer:
- create a fork (by eduardofilo or some shared organization account)
- then start implementing small fixes/changes

Perhaps I could help with some small fixes or changes. My C knowledge is basic and I have only one specific device to test on.

As for this specific change; it might not fit perfectly in some themes and I think it is too early for a "public" release. Chinese and Japanse characters work, but Korean is not supported in the adapted font as far as I can tell.

**Reply by eduardofilo** [reply]
I guess the experimental branch includes the changes of the preliminary version I installed in v1.3 that FGL82 passed to me privately, but that I had to remove in v1.3.1 because the bootloops cases increased among users. That version had very interesting things like the organization of `rom_preferences` in directories to avoid collisions between systems, but as I say it needs debugging.

I have already thought about forking that project to at least debug the problems related to the `last_state.sav` file, which are the most, but the image support is already taking the time I can dedicate to these topics.

**Reply by eduardofilo** [reply]
By the way, while checking the SimpleMenu forks I saw that there was already one dedicated to convert it (rather than implement) to Chinese. This is the only commit with changes.

https://github.com/lualiliu/simplemenuCN/commit/5439a3ab7a926ead794d175d40b3a4d4171612d2

**Reply by a-dekker** [reply]
I noticed as well. And there is another fork that addresses Chinese, but it only translates the UI and does not fix the rom names. And I doubt if the fork you mention does this as well, but I have not checked.

**Reply by bobdos** [reply]
hi @a-dekker is that possible to let adam team merge your changes into the official adam release?

**Reply by a-dekker** [reply]
Simplemenu is not managed within the adam-image project, and eduardofilo has no plans in that direction. So merging is not possible. Just adding some compiled binary without source code is not the way it should be managed in an 'official' project.
And as I mentioned before I do not want to create a fork and become the sole maintainer.

**Comment by pavanini** [comment]
Hola Eduardo, primero que nada gracias por tu trabajo, eso tu imagen casi desde que se empezó a publicar y me encanta, quisiera saber si es posible que revises un error en el emulador de neogeo donde si guado una partida al cargarla el juego ya no tiene sonido, por lo demás tu imagen esta muy buena, espero sigas con tu gran trabajo

**Reply by eduardofilo** [reply]
OK, lo revisaré. Gracias.

**Reply by eduardofilo** [reply]
Hola. Es un bug del core FBA 2012. Lo he reproducido en una RG503 con ArkOS seleccionando ese mismo core para emular NeoGeo. Lo único que puedes hacer aparte de esperar a que supuestamente algún día se solucione en futuras versiones, es utilizar la alternativa que hay para emular NeoGeo en la imagen, es decir el emulador standalone FBA. He comprobado que no sufre de ese problema.
